src/googmodule.ts

```diff
@@ -0,0 +1,388 @@
+  const [exp, esM, val] = expr.arguments;
+  if (!ts.isIdentifier(exp) || exp.escapedText !== 'exports') return false;
+  if (!ts.isStringLiteral(esM) || esM.text !== '__esModule') return false;
+  if (!ts.isObjectLiteralExpression(val) || val.properties.length !== 1) return false;
+  const prop = val.properties[0];
+  if (!ts.isPropertyAssignment(prop)) return false;
+  const ident = prop.name;
+  if (!ident || !ts.isIdentifier(ident) || ident.text !== 'value') return false;
+  return prop.initializer.kind === ts.SyntaxKind.TrueKeyword;
+}
+
+/**
+ * Returns the string argument if call is of the form
+ *   require('foo')
+ */
+function extractRequire(call: ts.CallExpression): string|null {
+  // Verify that the call is a call to require(...).
+  if (call.expression.kind !== ts.SyntaxKind.Identifier) return null;
+  const ident = call.expression as ts.Identifier;
+  if (ident.escapedText !== 'require') return null;
+
+  // Verify the call takes a single string argument and grab it.
+  if (call.arguments.length !== 1) return null;
+  const arg = call.arguments[0];
+  if (arg.kind !== ts.SyntaxKind.StringLiteral) return null;
+  return (arg as ts.StringLiteral).text;
+}
+
+function createGoogCall(methodName: string, literal: ts.StringLiteral): ts.CallExpression {
+  return ts.createCall(
+      ts.createPropertyAccess(ts.createIdentifier('goog'), methodName), undefined, [literal]);
+}
+
+/**
+ * Extracts the namespace part of a goog: import, or returns null if the given
+ * import is not a goog: import.
+ */
+export function extractGoogNamespaceImport(tsImport: string): string|null {
+  if (tsImport.match(/^goog:/)) return tsImport.substring('goog:'.length);
+  return null;
+}
+
+// Matches common extensions of TypeScript input filenames
+const TS_EXTENSIONS = /(\.ts|\.d\.ts|\.js|\.jsx|\.tsx)$/;
+
+/**
+ * Convert from implicit `import {} from 'pkg'` to `import {} from 'pkg/index'.
+ * TypeScript supports the shorthand, but not all ES6 module loaders do.
+ * Workaround for https://github.com/Microsoft/TypeScript/issues/12597
+ */
+export function resolveIndexShorthand(
+    host: {options: ts.CompilerOptions, host: ts.ModuleResolutionHost}, fileName: string,
+    imported: string): string {
+  const resolved = ts.resolveModuleName(imported, fileName, host.options, host.host);
+  if (!resolved || !resolved.resolvedModule) return imported;
+  const requestedModule = imported.replace(TS_EXTENSIONS, '');
+  const resolvedModule = resolved.resolvedModule.resolvedFileName.replace(TS_EXTENSIONS, '');
+  if (resolvedModule.indexOf('node_modules') === -1 &&
+      requestedModule.substr(requestedModule.lastIndexOf('/')) !==
+          resolvedModule.substr(resolvedModule.lastIndexOf('/'))) {
+    imported = './' + path.relative(path.dirname(fileName), resolvedModule).replace(path.sep, '/');
+  }
+  return imported;
+}
+
+/**
+ * importPathToGoogNamespace converts a TS/ES module './import/path' into a goog.module compatible
```

Comment:
Maybe mention that it also handles "goog:" imports, otherwise it looks redundant with pathToModuleName.